dart_canvas_parser/canvas_api.py

```diff
@@ -1,231 +1,96 @@
-    log.info("Getting profile")
-    users.get_user_profile(context, 'self').json()
-    log.info("Fetched profile")
-    return True
-
-
-def get_all_courses(user_model):
-    return _sdk_call(user_model, _get_all_courses)
-
-
-def _get_all_courses(context):
-    lst = get_all_list_data(
-        context,
-        enrollments.list_enrollments_users,
-        user_id='self',
-        state='completed',
-        type=['TeacherEnrollment', 'TaEnrollment', 'ObserverEnrollment',
-              'DesignerEnrollment'],
-    )
-    lst += get_all_list_data(
-        context,
-        enrollments.list_enrollments_users,
-        user_id='self',
-        state='active',
-        type=['TeacherEnrollment', 'TaEnrollment', 'ObserverEnrollment',
-              'DesignerEnrollment'],
-    )
-    course_list = []
-    for e in lst:
-        course_list.append(_get_course(context, e['course_id']))
-    return course_list
-
-
-def get_active_courses(user_model):
-    """
-    Makes an SDK call (_sdk_call transforms user model into canvas_python_sdk
-    request context)
-    :param user_model: instance of canvas_adapter.models.CanvasUser
-    :return: results of _get_active_courses()
-    """
-    return _sdk_call(user_model, _get_active_courses)
-
-
-def _get_active_courses(context):
-    """
-    Gets a full list of courses for which the user has a teacher or designer
-    role and which are either unpublished or available (i.e. not completed or
-    deleted).
-    :param context: request context required by canvas_python_sdk
-    :return: list of dicts representing Canvas courses (see
-    https://canvas.instructure.com/doc/api/courses.html#Course)
-    """
-    lst = get_all_list_data(
-        context,
-        courses.list_your_courses,
-        include='term',
-        state=['unpublished', 'available']
-    )
-    course_list = []
-    for c in lst:
-        for e in c['enrollments']:
-            if e['type'] in ['teacher', 'designer']:
-                course_list.append(c)
-                break
-    log.info("Found {} active courses.".format(len(course_list)))
-    return course_list
-
-
-def get_course(user_model, course_id):
-    return _sdk_call(user_model, _get_course, course_id=course_id)
-
-
-def _get_course(context, course_id):
-    url = "{}/v1/courses/{}?include[]=syllabus_body&include[]=term".format(
-        context.base_api_url, course_id
-    )
-    response = requests.get(
-        url, headers={"Authorization": "Bearer {}".format(context.auth_token)}
-    )
-    if response.status_code != 200:
-        if response.status_code == 401 and 'WWW-Authenticate' in response.headers:
-            raise InvalidOAuthTokenError(
-                "OAuth Token used to make request to %s is invalid" % response.url)
-        raise CanvasAPIError(
-            status_code=response.status_code,
-            msg=unicode(response.json()),
-            error_json=response.json(),
-        )
-    return response.json()
 
+class CanvasApi(object):
```

Comment:
I like having a `CanvasApi` layer. I would prefer to swap out some of our custom code for an existing Python Canvas API, and this `CanvasApi` wrapper allows us to do that pretty easily I'd hope.
See:
- https://github.com/penzance/canvas_python_sdk
- https://github.com/ucfopen/canvasapi

Reply:
It is actually already wrapping methods from the penzance/canvas_python_sdk! Maybe having the wrapper would make it easier to switch it out for the ucfopen library if it made sense to